Set idle socket keepalive time to 4 seconds

ghoshkaj opened the pull request:
The tinylicious tests infrequently but consistently fail with 502 error code. A possible cause might be because the node (Axios) server socket timeout is 15s, whereas the underlying socket timeout default timeout is 5s. This PR switches to using agentkeepalive package that forces the Node server to understand and close connections idle sockets after 4 seconds so that Node doesn't try to use an already closed socket. 

Implementing this fix on both client calls and internal server calls, since both locations can have this problem.

ghoshkaj committed:
set keepalive to 4

curtisman commented on server/tinylicious/src/index.ts:
```diff
@@ -20,7 +20,7 @@ import { TinyliciousRunnerFactory } from "./runnerFactory";
 // keep the TCP connection open so that they can be reused
 // TODO: Set this globally since the Historian use the global Axios default instance.  Make this encapsulated.
-Axios.defaults.httpAgent = new http.Agent({ keepAlive: true });
+Axios.defaults.httpAgent = new http.Agent({ keepAlive: true, timeout: (60 * 1000) * 4 });
```
I don't think `timeout` is not related to keepAlive, it is just a general timeout for the socket.
I beileve you will need to use [agentkeepalive](https://www.npmjs.com/package/agentkeepalive) package, which has an extra option to specify `freeSocketTimeout`.   I don't think we need to specify a number before it is already default 4000ms.

ghoshkaj committed:
use agentkeepalive packages

ghoshkaj committed:
Merge branch 'main' into set-keepalive-timout-4s

ghoshkaj committed:
package-lock.json

alexvy86 committed:
Remove references to deleted file (#11607)

alexvy86 committed:
Basic SharedMap and SharedDirectory memory tests (#11299)

alexvy86 committed:
Sharedstring memory tests (#11403)

Josmithr committed:
Add `@fluidframework/counter` API docs to website nav (#11635)

![image](https://user-images.githubusercontent.com/54606601/186233989-4457b1f1-6a70-4711-b888-2c455fd2974c.png)

Also update package to use `strict` API-Extractor configuration, and updated some comments.

ghoshkaj committed:
use agentkeepalive for client calls as well

ghoshkaj commented:
Sadly, I don't think this is fixing the error after all. 4/10 runs failed:
Failed runs:

[Pipelines - Run 87750 (azure.com)](https://dev.azure.com/fluidframework/internal/_build/results?buildId=87750&view=ms.vss-test-web.build-test-results-tab&runId=2002920&resultId=103456&paneView=debug)
[Pipelines - Run 87771 (azure.com)](https://dev.azure.com/fluidframework/internal/_build/results?buildId=87771&view=ms.vss-test-web.build-test-results-tab&runId=2003778&resultId=103980&paneView=debug)
[Pipelines - Run 87772 (azure.com)](https://dev.azure.com/fluidframework/internal/_build/results?buildId=87772&view=ms.vss-test-web.build-test-results-tab&runId=2003596&resultId=101971&paneView=debug)
[Pipelines - Run 87776 (azure.com)](https://dev.azure.com/fluidframework/internal/_build/results?buildId=87776&view=ms.vss-test-web.build-test-results-tab&runId=2003962&resultId=103178&paneView=debug)

ghoshkaj committed:
Revert "package-lock.json"

This reverts commit 4fdd613a1d699693afa71bb93d80ebe5c1aca013.

ghoshkaj committed:
Revert "Revert "package-lock.json""

This reverts commit 4c70ba86599846f21ee5ea6bd65a88033bd87e2d.

ghoshkaj committed:
Merge branch 'set-keepalive-timout-4s' of github.com:ghoshkaj/FluidFramework into set-keepalive-timout-4s

ghoshkaj committed:
Revert "use agentkeepalive for client calls as well"

This reverts commit 2e73072262ac35680c74f89b7a8133b50ade8b6b.
